clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -457,6 +459,13 @@ public static ByteBuffer toBuffer(Struct struct) {
         return tps;
     }
 
+    public static <T extends Throwable> T assertFutureThrows(KafkaFuture<?> future, Class<T> exceptionClass) {
```

Comment:
Nit: shall we call the parameter `exceptionCauseClass` and add brief javadoc saying that this checks the cause of the execution exception?